File: src/main/java/com/owiseman/dataapi/plugins/sdk/CommandResult.java
package com.owiseman.dataapi.plugins.sdk;

/**
 * 命令执行结果类
 */
public class CommandResult {
    private boolean success;
    private String result;
    private String errorMessage;

    public CommandResult(boolean success, String result) {
        this.success = success;
        this.result = result;
    }

    public CommandResult(boolean success, String result, String errorMessage) {
        this.success = success;
        this.result = result;
        this.errorMessage = errorMessage;
    }

    public static CommandResult success(String result) {
        return new CommandResult(true, result);
    }

    public static CommandResult error(String errorMessage) {
        return new CommandResult(false, null, errorMessage);
    }

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getResult() {
        return result;
    }

    public void setResult(String result) {
        this.result = result;
    }

    public String getErrorMessage() {
        return errorMessage;
    }

    public void setErrorMessage(String errorMessage) {
        this.errorMessage = errorMessage;
    }
}

File: python/command_result.py
class CommandResult:
    """命令执行结果类"""
    
    def __init__(self, success=False, result=None, error_message=None):
        self.success = success
        self.result = result
        self.error_message = error_message
        
    def is_success(self):
        return self.success
    
    def set_success(self, success):
        self.success = success
        
    def get_result(self):
        return self.result
    
    def set_result(self, result):
        self.result = result
        
    def get_error_message(self):
        return self.error_message
    
    def set_error_message(self, error_message):
        self.error_message = error_message